src/validator/CreditCardValidator.java

```diff
@@ -0,0 +1,62 @@
+package validator;
+
+import java.util.ArrayList;
+
+public class CreditCardValidator implements CreditCardChecking {
+
+	@Override
+	public boolean checkValidity(CreditCardVendor ccv, String number) {
+		ArrayList<Integer> chars = ccv.getNumberOfChars();
+		ArrayList<String> masks = ccv.getIIN();
+		
+		boolean result1 = checkLengthCorrectness(number, chars);
+		boolean result2 = checkIINMaskCorrectness (number, masks);
+		boolean result3 = checkLuhnalgorithmCorrectness(number);
+		
+		return result1 && result2 && result3;
+	}
+	
+	private boolean checkLengthCorrectness(String number, ArrayList<Integer> chars){
+		boolean result = false;
+		for (int i = 0; i<chars.size();i++){
+			if (chars.get(i) == number.length())
+				result = true;
+		}
+		return result;
+	}
+	
+	private boolean checkIINMaskCorrectness (String number, ArrayList<String> masks){
+		boolean result = false;
+		for (int i = 0; i<masks.size();i++){
+			if (number.startsWith(masks.get(i)))
+				result = true;
+		}
+		return result;
+	}
+	
+	private boolean checkLuhnalgorithmCorrectness(String number){
+		boolean result = false;
+		
+		String[] characters = number.split("");
+		int[] intCharsReverse = new int[characters.length];
+		int sum = 0;
+		
+		for(int i = 0;i<characters.length;i++){
+			intCharsReverse[characters.length-1-i] = Integer.parseInt(characters[i]);
```

Comment:
parseInt throws NumericFormatException if the card number string to be verified contains non-numeric characters. 
The exception is caught by the CreditCardValidatorProject class and an error message is printed. Using exceptions for flow control is not recommended because of the performance overhead, which can make the application more vulnerable to DDoS attacks. A better alternative is to detect non-numeric characters and set a flag which can be read by the calling class, for example:
```suggestion
private boolean isFormatValid;
...
if (characters[i].length() < 1 || characters[i].charAt(0) < '0' || characters[i].charAt(0) > '9') {
  isFormatValid = false;
  return false;
} else {
  intCharsReverse[characters.length - 1 - i] = characters[i].charAt(0) - '0';
}
```
If the application is converted to a web service such attempts should be logged to detect potential DDoS sources.